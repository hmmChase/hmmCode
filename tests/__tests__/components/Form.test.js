import { render, prettyDOM, fireEvent } from '@testing-library/react';
import Form from '../../components/Form';

describe(Form, () => {
  it('matches snapshot', () => {
    const mockProps = {};

    const utils = render(<Form {...mockProps} />);

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
  });

  it('updates value on keypress', () => {
    const mockProps = {};

    const utils = render(<Form {...mockProps} />);

    const input1 = utils.getByLabelText('Label :');

    const enterEvent = { which: 13, charCode: 13 };

    fireEvent.keyPress(input1, enterEvent);

    const input2 = utils.getByLabelText('Label :');

    expect(input2.value).toEqual('keyPress 13');
  });

  it('updates value on change', () => {
    const mockProps = {};

    const utils = render(<Form {...mockProps} />);

    const input1 = utils.getByLabelText('Label :');

    const changeEvent = { target: { value: 'test' } };

    fireEvent.change(input1, changeEvent);

    const input2 = utils.getByLabelText('Label :');

    expect(input2.value).toEqual('test');
  });
});
