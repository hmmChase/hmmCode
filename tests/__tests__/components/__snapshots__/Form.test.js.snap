// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
<form>
  <label
    for="input-text"
  >
    Label :
  </label>
  <input
    id="input-text"
    type="text"
    value=""
  />
</form>
`;
